test/e2e/operator_test.go

```diff
@@ -869,6 +871,158 @@ func TestRouteAdmissionPolicy(t *testing.T) {
 	}
 }
 
+func TestSyslog(t *testing.T) {
+	// Set up rsyslog.
+	syslogPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-ingress",
+			Name:      "syslog",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "syslog",
+					Image: "openshift/origin-haproxy-router:v4.0",
+					Command: []string{
+						"/sbin/rsyslogd", "-n",
+						"-i", "/tmp/rsyslog.pid",
+						"-f", "/etc/rsyslog/rsyslog.conf",
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "rsyslog-config",
+							MountPath: "/etc/rsyslog",
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "rsyslog-config",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "rsyslog-conf",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if err := kclient.Create(context.TODO(), syslogPod); err != nil {
+		t.Fatalf("failed to create pod for rsyslog: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), syslogPod); err != nil {
+			t.Fatalf("failed to delete pod %s: %v", syslogPod.Name, err)
+		}
+	}()
+	syslogConfigmap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rsyslog-conf",
+			Namespace: "openshift-ingress",
+		},
+		Data: map[string]string{
+			"rsyslog.conf": `$ModLoad imudp
+$UDPServerRun 10514
+$ModLoad omstdout.so
+*.* :omstdout:
+`,
+		},
+	}
+	if err := kclient.Create(context.TODO(), syslogConfigmap); err != nil {
+		t.Fatalf("failed to create configmap for rsyslog: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), syslogConfigmap); err != nil {
+			t.Fatalf("failed to delete configmap %s: %v", syslogConfigmap.Name, err)
+		}
+	}()
+
+	// Get the rsyslog endpoint.
+	var syslogAddress string
+	err := wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), types.NamespacedName{Namespace: syslogPod.Namespace, Name: syslogPod.Name}, syslogPod); err != nil {
+			return false, nil
+		}
+		syslogAddress = syslogPod.Status.PodIP
+		if len(syslogAddress) == 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to observe syslog pod IP address: %v", err)
+	}
+
+	// Create an ingresscontroller that logs to the endpoint.
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "syslog"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
```

Comment:
Do we need something even more lightweight than newPrivateController? For example, if you used NodePort, I think the test would be just as good but wouldn't consume a costly LB, DNS entries, etc

Reply:
`newPrivateController` uses the "Private" endpoint publishing strategy type, so it doesn't do LB or DNS.

Reply:
Whoops, confused it with internal. Nevermind.